app/views/stories/_main_stories_feed.html.erb

```diff
@@ -19,3 +19,83 @@
     <div class="single-article-small-pic" id="article-index-hidden-div" style="display:none"></div>
   <% end %>
 </div>
+
+<script defer>
```

Comment:
syntax (blocking): Avoid inline JavaScript in ERB templates for a few reasons:

* One it is inline and won't be cached in the browser (well potentially, but as part of an HTML fragment in our server-side caching)
* This code can not be tested via unit/integration tests in Jest

Prefer to create a webpacker pack file, which can import most the required functions and run them. See the Forem [webpacker](https://docs.forem.com/frontend/webpacker/#webpacker) documentation as well as the webpacker [folder structure](https://github.com/rails/webpacker/blob/master/docs/folder-structure.md) documentation.

Reply:
Wow! completely missed this.

Refactored to reuse the code of the list navigation list, but with different values, since the logged out feed is different. It felt so good to remove that specific unreusable code 👍